### PR TITLE
add createReadStream, createWriteStream #4

### PR DESCRIPTION
**Pull request by coderaiser**
Added `createReadStream` and `createWriteStream` with help of a [dropbox-stream](https://www.npmjs.com/package/dropbox-stream).

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -223,6 +224,35 @@ export default ({ apiKey = null, client = null } = {}) => {
...
+         *
+         * @param {String} token
+         * @param {String} remotePath
+         * @returns {Stream}
+         */
+        createWriteStream(filepath) {
+            return dropboxStream.createDropboxUploadStream({
+                token: client.accessToken,
+                filepath,
+                chunkSize: 1000 * 1024
```
Comment (sallar) [inline src/index.js]: Maybe allow setting the `chunkSize` through an `options` object?

**Comment by sallar** [comment]
Ah I think you havent updated `readOnly`

File: src/index.js [hunk 2]
```diff
@@ -223,6 +224,35 @@ export default ({ apiKey = null, client = null } = {}) => {
...
+        },
+
+        /**
+         * create read stream
+         *
+         * @param {String} token
+         * @param {String} remotePath
+         * @returns {Stream}
+         */
+        createReadStream(filepath) {
```
Comment (piranna) [inline src/index.js]: Could be added the `options` argument, speciall the `start` and `end` fields?

Comment (sallar) [reply src/index.js]: yup!

File: src/index.js (hunk 1 above)
Comment (coderaiser) [reply src/index.js]: [fs.createWriteStream](https://nodejs.org/dist/latest-v9.x/docs/api/fs.html#fs_fs_createwritestream_path_options) doesn’t supports `chunkSize`

File: src/index.js (hunk 2 above)
Comment (coderaiser) [reply src/index.js]: `dropbox-stream` doesn’t support `start` and `end`

File: src/index.js (hunk 1 above)
Comment (piranna) [reply src/index.js]: Doesn't forbid us to allow it as an option too...

File: src/index.js (hunk 2 above)
Comment (piranna) [reply src/index.js]: Then time to add support for it :-)

Comment (perry-mitchell) [reply src/index.js]: I would personally suggest to add those niceties in a separate PR to not block this one. I would argue that their usage is uncommon.

Comment (sallar) [reply src/index.js]: Yup Im going to merge this :) thanks all

Comment (piranna) [reply src/index.js]: > I would personally suggest to add those niceties in a separate PR to not block this one

Ok, I agree.

**Comment by sallar** [comment]
Thanks @coderaiser I merged this and I will update and add the rest of discussed stuff myself before releasing it. Therefore #4 is still open until I close it with my additions. Cheers

**Pull request merged**